pkg/app/web/src/components/deployment-frequency-chart.tsx

```diff
@@ -12,26 +12,36 @@ import {
 } from "recharts";
 import dayjs from "dayjs";
 import { theme } from "../theme";
+import { WarningOutlined } from "@material-ui/icons";
 
 const useStyles = makeStyles((theme) => ({
   root: {
     padding: theme.spacing(3),
     minWidth: 600,
   },
+  noDataMessage: {
+    display: "flex",
+  },
+  noDataMessageIcon: {
+    marginRight: theme.spacing(1),
+  },
 }));
 
 interface Props {
   data: InsightDataPoint.AsObject[];
 }
 
+const tickFormatter = (time: number | string): string =>
+  dayjs(time).format("MMM DD");
+
+const labelFormatter = (time: number | string): string =>
+  dayjs(time).format("YYYY MMM DD");
+
+const NO_DATA_TEXT = "No data is available.";
```

Comment:
should move to constant directory 👀

Reply:
in the second thought, I guess this value is encapsulated with this component only, so the current place is LGTM 👍